src/crimson/os/seastore/cache.h

```diff
@@ -181,7 +181,7 @@ class Cache {
       auto ret = TCachedExtentRef<T>(static_cast<T*>(&*iter));
       return ret->wait_io(
       ).then([ret=std::move(ret)]() mutable -> get_extent_ret<T> {
-	assert(ret->is_valid());
+	// ret may be invalid, caller must check
```

Comment:
so onode tree and omap tree still need check in every continuation if node/parent nod is valid?

Reply:
This method is not exposed through TransactionManager, it's internal.  Functions that return interruptible_futures and take a transaction as an argument should implictely check that the transaction is still valid.  That said, until TransactionManager users are converted to use interruptible_future, they will still need to check that the transaction hasn't conflicted.